README.md

```diff
@@ -37,6 +38,49 @@ This package is a work in progress:
 - JavaScript/TypeScript package, CLI
 - TODO: API and web UI
 
+### Provider and Model Matching
```

Comment:
I don't think we need this section in the main readme - we don't cover the rest of the mechanics in this much detail.

The one line above is sufficient.